cpp/include/raft/neighbors/detail/cagra/utils.hpp

```diff
@@ -150,4 +152,93 @@ struct gen_index_msb_1_mask {
 };
 }  // namespace utils
 
+/**
+ * Utility to sync memory from a host_matrix_view to a device_matrix_view
+ *
+ * In certain situations (UVM/HMM/ATS) host memory might be directly accessible on the
+ * device, and no extra allocations need to be performed. This class checks
+ * if the host_matrix_view is already accessible on the device, and only creates device
+ * memory and copies over if necessary. In memory limited situations this is preferable
+ * to having both a host and device copy
+ */
+template <typename T, typename IdxT>
+class device_matrix_view_from_host {
+ public:
```

Comment:
I'm definitely okay keeping this as an internal utility for now. Could you add a todo to the docs here (and for the host->device conversion function) to use `mdbuffer` for this once it's available?

Reply:
added a TODO here https://github.com/rapidsai/raft/pull/1790/commits/6552c66c820431285c5ee01705dbfaaf7b4435ed